doc/source/train/examples/lightning/lightning_mnist_example.ipynb

```diff
@@ -690,51 +970,158 @@
+      "\u001b[2m\u001b[36m(TestWorker pid=20600)\u001b[0m 2023-06-13 16:06:07.708119: I tensorflow/core/util/port.cc:104] oneDNN custom operations are on. You may see slightly different numerical results due to floating-point round-off errors from different computation orders. To turn them off, set the environment variable `TF_ENABLE_ONEDNN_OPTS=0`.\n"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Testing DataLoader 0: 100%|██████████| 20/20 [00:00<00:00, 22.10it/s]\n"
+     ]
+    },
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "\u001b[2m\u001b[36m(TestWorker pid=20600)\u001b[0m 2023-06-13 16:06:08.680418: W tensorflow/compiler/xla/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libnvinfer.so.7'; dlerror: libnvinfer.so.7: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/nvidia/lib:/usr/local/nvidia/lib64\n",
+      "\u001b[2m\u001b[36m(TestWorker pid=20600)\u001b[0m 2023-06-13 16:06:08.680524: W tensorflow/compiler/xla/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libnvinfer_plugin.so.7'; dlerror: libnvinfer_plugin.so.7: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/nvidia/lib:/usr/local/nvidia/lib64\n",
+      "\u001b[2m\u001b[36m(TestWorker pid=20600)\u001b[0m 2023-06-13 16:06:08.680532: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Cannot dlopen some TensorRT libraries. If you would like to use Nvidia GPU with TensorRT, please make sure the missing libraries mentioned above are installed properly.\n"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Testing DataLoader 0: 100%|██████████| 20/20 [00:02<00:00,  7.31it/s]\n",
+      "\u001b[2m\u001b[36m(TestWorker pid=20600)\u001b[0m ┏━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━┓\n",
+      "\u001b[2m\u001b[36m(TestWorker pid=20600)\u001b[0m ┃        Test metric        ┃       DataLoader 0        ┃\n",
+      "\u001b[2m\u001b[36m(TestWorker pid=20600)\u001b[0m ┡━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━┩\n",
+      "\u001b[2m\u001b[36m(TestWorker pid=20600)\u001b[0m │       test_accuracy       │    0.9740999937057495     │\n",
+      "\u001b[2m\u001b[36m(TestWorker pid=20600)\u001b[0m └───────────────────────────┴───────────────────────────┘\n"
      ]
     }
    ],
    "source": [
-    "from ray.train.lightning import LightningPredictor\n",
+    "import ray\n",
+    "import pytorch_lightning as pl\n",
     "\n",
-    "predictor = LightningPredictor.from_checkpoint(\n",
-    "    checkpoint, MNISTClassifier, use_gpu=use_gpu\n",
+    "from pytorch_lightning.plugins.environments.lightning_environment import (\n",
+    "    LightningEnvironment,\n",
     ")\n",
+    "from ray.air.util.torch_dist import (\n",
+    "    TorchDistributedWorker,\n",
```

Comment:
is this a public API now @matthewdeng ?

Reply:
It is not at the moment. Nothing has changed since it was [introduced](https://github.com/ray-project/ray/pull/34202).

Reply:
Actually, we are already using it in the [OPT-30B Batch Inference](https://docs.ray.io/en/master/ray-air/examples/opt_deepspeed_batch_inference.html?highlight=OPT%20#define-a-deepspeed-predictor) example.